hyperspy/tests/axes/test_axes_manager.py

```diff
@@ -75,3 +77,102 @@ def test_update_from(self):
+        signal_axis = s.axes_manager.signal_axes[0]
+        signal_extent = (signal_axis.low_value, signal_axis.high_value)
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 0
+        assert () == s.axes_manager.navigation_extent
+
+    def test_2d_signal1d(self):
+        s = Signal1D(arange(100).reshape(10, 10))
+        assert len(s.axes_manager.signal_extent) == 2
+        signal_axis = s.axes_manager.signal_axes[0]
+        signal_extent = (signal_axis.low_value, signal_axis.high_value)
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 2
+        nav_axis = s.axes_manager.navigation_axes[0]
+        nav_extent = (nav_axis.low_value, nav_axis.high_value)
+        assert nav_extent == s.axes_manager.navigation_extent
+
+    def test_3d_signal1d(self):
+        s = Signal1D(arange(1000).reshape(10, 10, 10))
+        assert len(s.axes_manager.signal_extent) == 2
+        signal_axis = s.axes_manager.signal_axes[0]
+        signal_extent = (signal_axis.low_value, signal_axis.high_value)
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 4
+        nav_axis0 = s.axes_manager.navigation_axes[0]
+        nav_axis1 = s.axes_manager.navigation_axes[1]
+        nav_extent = (
+                nav_axis0.low_value, nav_axis0.high_value,
+                nav_axis1.low_value, nav_axis1.high_value,
+                )
+        assert nav_extent == s.axes_manager.navigation_extent
+
+    def test_2d_signal2d(self):
+        s = Signal2D(arange(100).reshape(10, 10))
+        assert len(s.axes_manager.signal_extent) == 4
+        signal_axis0 = s.axes_manager.signal_axes[0]
+        signal_axis1 = s.axes_manager.signal_axes[1]
+        signal_extent = (
+                signal_axis0.low_value, signal_axis0.high_value,
+                signal_axis1.low_value, signal_axis1.high_value,
+                )
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 0
+        assert () == s.axes_manager.navigation_extent
+
+    def test_3d_signal2d(self):
+        s = Signal2D(arange(1000).reshape(10, 10, 10))
+        assert len(s.axes_manager.signal_extent) == 4
+        signal_axis0 = s.axes_manager.signal_axes[0]
+        signal_axis1 = s.axes_manager.signal_axes[1]
+        signal_extent = (
+                signal_axis0.low_value, signal_axis0.high_value,
+                signal_axis1.low_value, signal_axis1.high_value,
+                )
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 2
+        nav_axis = s.axes_manager.navigation_axes[0]
+        nav_extent = (nav_axis.low_value, nav_axis.high_value)
+        assert nav_extent == s.axes_manager.navigation_extent
+
+    def test_changing_scale_offset(self):
```

Comment:
All looks good, but this last test. It only ever references `high_value` and `low_value` of each axis, and does not actually test that changing scale or offset changes this.

I think it best if instead of this you'd include a general `DataAxis` test that checks if `low_value` and `high_value` are updated appropriately when changing `offset` and `scale`